gateway/internal/petition/petition_repository.go

```diff
@@ -27,17 +30,83 @@ type petitionRepository struct {
 	client pb.PetitionServiceClient
 }
 
-func (svc *petitionRepository) Get() {
+func mapPetition(pbPetition *pb.Petition) model.Petition {
+	var petition model.Petition
+
+	petition.Title = pbPetition.Title
+	petition.Category = pbPetition.Category
+	petition.Description = pbPetition.Description
+	petition.Image = pbPetition.Image
+	petition.Status = uint32(pbPetition.Status)
+	petition.UserID = uint(pbPetition.UserId)
+
+	return petition
 }
 
-func (svc *petitionRepository) Delete() {
+func (repo *petitionRepository) CreatePetition(petition model.Petition) (uint32, error) {
+	resp, err := repo.client.CreatePetition(context.Background(), &pb.CreatePetitionRequest{
+		Petition: &pb.Petition{
+			PetitionId: uint32(petition.PetitionId),
+			Title:      petition.Title,
+			Category:   petition.Category,
+			Image:      petition.Image,
+			Status:     uint32(petition.Status),
+			UserId:     uint32(petition.UserID),
+		},
+	})
+
+	if err != nil {
+		return 0, err
+	}
+	return resp.PetitionId, nil
+}
+
+func (repo *petitionRepository) GetPetitions(query model.PaginationQuery) ([]model.Petition, error) {
```

Comment:
Why not just get the petition id from the URL? Why are you binding a whole petition from the request body?